src/backend/marsha/core/api.py

```diff
@@ -673,24 +673,45 @@ def get_permissions(self):
         if self.action == "metadata":
             permission_classes = [permissions.IsVideoToken]
         elif self.action in ["create", "list"]:
-            # NB: list should be restricted by queryset, not resource id
             permission_classes = [
-                permissions.IsTokenInstructor | permissions.IsTokenAdmin
+                permissions.IsTokenInstructor
+                | permissions.IsTokenAdmin
+                | permissions.IsParamsVideoAdminThroughOrganization
+                | permissions.IsParamsVideoAdminThroughPlaylist
             ]
         else:
             permission_classes = [
                 permissions.IsTokenResourceRouteObjectRelatedVideo
                 & permissions.IsTokenInstructor
                 | permissions.IsTokenResourceRouteObjectRelatedVideo
                 & permissions.IsTokenAdmin
+                | permissions.IsRelatedVideoPlaylistAdmin
+                | permissions.IsRelatedVideoOrganizationAdmin
             ]
         return [permission() for permission in permission_classes]
 
     def get_queryset(self):
         """Restrict list access to timed text tracks related to the video in the JWT token."""
         user = self.request.user
-        if isinstance(user, TokenUser):
+        if isinstance(user, TokenUser) and not user.token.get("user_id"):
             return TimedTextTrack.objects.filter(video__id=user.id)
+
+        if isinstance(user, TokenUser) and user.token.get("user_id") == user.token.get(
+            "resource_id"
+        ):
+            return TimedTextTrack.objects.filter(
+                Q(
+                    video__playlist__user_accesses__in=models.PlaylistAccess.objects.filter(
+                        user__id=user.token.get("user_id")
+                    )
+                )
+                | Q(
+                    video__playlist__organization__user_accesses__in=models.OrganizationAccess.objects.filter(
+                        user__id=user.token.get("user_id")
+                    )
+                )
```

Comment:
I'm afraid this splits permission logic in 2 and will be harder to follow than grouping all permission related filtering at the same place.

Reply:
I updated it to change it from permission logic in `get_queryset` to simple filtering in `list`. IMO it's more correct and way simpler (and makes fewer queries too)